src/devices/laserFromDepth/laserFromDepth.cpp

```diff
@@ -263,38 +263,43 @@ void LaserFromDepth::run()
 
 
     float* pointer = (float*)m_depth_image.getPixelAddress(0, m_depth_height / 2);
-    
+    double angle, distance, infinity, angleShift;
+    size_t i;
+
+    infinity   = std::numeric_limits<double>::infinity();
+    angleShift = m_sensorsNum * m_resolution / 2;
+
     for (int elem = 0; elem < m_sensorsNum; elem++)
     {
-        double distance = *(pointer + elem); //m
-        double angle = elem*m_resolution; //deg
 
-        if (m_clip_min_enable)
+        angle    = elem * m_resolution;    //deg
+
+        //the 1 / cos(blabla) distortion simulate the way RGBD devices calculate the distance..
+        //sorry for the one line unreadable operation, it's for performance reason..
```

Comment:
This is not true.
In release mode, the line:
`distance = *(pointer + elem) * 1 /cos((angle - angleShift) * DEG2RAD); //m`
produces this asm code:

```
; 279  :         distance = *(pointer + elem) * 1 /cos((angle - angleShift) * DEG2RAD); //m

	cvtps2pd xmm0, xmm0
	mulsd	xmm1, QWORD PTR [edi+128]
	movsd	QWORD PTR _distance$1$[ebp], xmm0
	movaps	xmm0, xmm1
	movsd	QWORD PTR _angle$1$[ebp], xmm1
	subsd	xmm0, xmm2
	mulsd	xmm0, QWORD PTR __real@400921fb54442d18
	divsd	xmm0, QWORD PTR __real@4066800000000000
	call	__libm_sse2_cos_precise
```
while:
```
        distance = *(pointer + elem);
        distance = distance / cos((angle - angleShift) * DEG2RAD); //m
```
produces the same, identical, asm code:
```
; 279  :         distance = *(pointer + elem);

	cvtps2pd xmm0, xmm0
	mulsd	xmm1, QWORD PTR [edi+128]
	movsd	QWORD PTR _distance$1$[ebp], xmm0

; 280  :         distance = distance / cos((angle - angleShift) * DEG2RAD); //m

	movaps	xmm0, xmm1
	movsd	QWORD PTR _angle$1$[ebp], xmm1
	subsd	xmm0, xmm2
	mulsd	xmm0, QWORD PTR __real@400921fb54442d18
	divsd	xmm0, QWORD PTR __real@4066800000000000
	call	__libm_sse2_cos_precise
```
This because in Release mode a smart compiler produces optimizations which modify the execution order in a way which can be hardly predicted by a human.   
Obviously, in Debug mode, the two codes will be different, but in Debug mode we are pursuing readability, not performance.
Thus, a good programmer should never try to substitute himself to the compiler and always seek for a good compromise between clearness and conciseness.